python/cudf/cudf/io/csv.py

```diff
@@ -153,3 +155,7 @@ def to_csv(
             line_terminator=line_terminator,
             rows_per_chunk=rows_per_chunk,
         )
+
+    if path is None:
```

Comment:
I recommend using explicit guard to make it more readable.
```suggestion
    if isinstance(path_or_buf, StringIO)
```

Reply:
We'd actually want to extract and return csv as `string` from `StringIO` when path is None. But `path` can also be a `StringIO` object. In which case we just want to write to the `StringIO` buffer and not return anything. 

For example:

```python
>>> import cudf
>>> df = cudf.DataFrame({'a':[1, 2, 3]})
>>> df.to_csv()     #path is None
',a\n0,1\n1,2\n2,3\n'


>>> import io
>>> f = io.StringIO()
>>> df.to_csv(f)      #path is StringIO
>>> f.seek(0)
0
>>> f.read()
',a\n0,1\n1,2\n2,3\n'
```

Reply:
Ah I see the point. We want to return something only when `path=None`.

BTW, I think the docstrings should be updated to reflect that use case.

Reply:
Thanks for the catch @isVoid, updated the docs.